libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -1271,120 +1265,107 @@ instance Cass.Cql FeatureStatus where
-    afcMlsMigration :: f MlsMigrationConfig,
-    afcEnforceFileDownloadLocation :: f EnforceFileDownloadLocationConfig,
-    afcLimitedEventFanout :: f LimitedEventFanoutConfig
-  }
-
-type AllFeatureConfigs = AllFeatures WithStatus
+-- | list of available features config types
+type Features :: [Type]
+type Features =
+  [ LegalholdConfig,
+    SSOConfig,
+    SearchVisibilityAvailableConfig,
+    SearchVisibilityInboundConfig,
+    ValidateSAMLEmailsConfig,
+    DigitalSignaturesConfig,
+    AppLockConfig,
+    FileSharingConfig,
+    ClassifiedDomainsConfig,
+    ConferenceCallingConfig,
+    SelfDeletingMessagesConfig,
+    GuestLinksConfig,
+    SndFactorPasswordChallengeConfig,
+    MLSConfig,
+    ExposeInvitationURLsToTeamAdminConfig,
+    OutlookCalIntegrationConfig,
+    MlsE2EIdConfig,
+    MlsMigrationConfig,
+    EnforceFileDownloadLocationConfig,
+    LimitedEventFanoutConfig
+  ]
+
+-- | list of available features as a record
+type AllFeatures f = NP f Features
+
+-- | 'AllFeatures' specialised to the 'LockableFeature' functor
+type AllFeatureConfigs = AllFeatures LockableFeature
+
+-- | constraint synonym requiring the @c@ instance for the @f@ type constructor applied to type @a@ to hold
+class (c (f a)) => LiftForF c f a
+
+instance (c (f a)) => LiftForF c f a
+
+type LockableFeatureDefault = LiftForF Default LockableFeature
 
 instance Default AllFeatureConfigs where
-  def =
-    AllFeatures
-      { afcLegalholdStatus = defFeatureStatus,
-        afcSSOStatus = defFeatureStatus,
-        afcTeamSearchVisibilityAvailable = defFeatureStatus,
-        afcSearchVisibilityInboundConfig = defFeatureStatus,
-        afcValidateSAMLEmails = defFeatureStatus,
-        afcDigitalSignatures = defFeatureStatus,
-        afcAppLock = defFeatureStatus,
-        afcFileSharing = defFeatureStatus,
-        afcClassifiedDomains = defFeatureStatus,
-        afcConferenceCalling = defFeatureStatus,
-        afcSelfDeletingMessages = defFeatureStatus,
-        afcGuestLink = defFeatureStatus,
-        afcSndFactorPasswordChallenge = defFeatureStatus,
-        afcMLS = defFeatureStatus,
-        afcExposeInvitationURLsToTeamAdmin = defFeatureStatus,
-        afcOutlookCalIntegration = defFeatureStatus,
-        afcMlsE2EId = defFeatureStatus,
-        afcMlsMigration = defFeatureStatus,
-        afcEnforceFileDownloadLocation = defFeatureStatus,
-        afcLimitedEventFanout = defFeatureStatus
-      }
+  def = hcpure (Proxy @LockableFeatureDefault) def
+
+-- | object schema for nary products
+--
+-- TODO(mangoiv): generalize this to be useable with schema profunctor
+class HObjectSchema c xs where
+  hobjectSchema :: (forall cfg. (c cfg) => ObjectSchema SwaggerDoc (f cfg)) -> ObjectSchema SwaggerDoc (NP f xs)
+
+instance HObjectSchema c '[] where
+  hobjectSchema _ = pure Nil
+
+instance (HObjectSchema c xs, c x) => HObjectSchema c ((x :: Type) : xs) where
+  hobjectSchema f = (:*) <$> hd .= f <*> tl .= hobjectSchema @c @xs f
```

Comment:
This should probably be moved to `schema-profunctor` or a corresponding utility module.
```suggestion
-- | object schema for nary products
class HObjectSchema c xs where
  hobjectSchema :: (forall cfg. (c cfg) => ObjectSchema SwaggerDoc (f cfg)) -> ObjectSchema SwaggerDoc (NP f xs)

instance HObjectSchema c '[] where
  hobjectSchema _ = pure Nil

instance (HObjectSchema c xs, c x) => HObjectSchema c ((x :: Type) : xs) where
  hobjectSchema f = (:*) <$> hd .= f <*> tl .= hobjectSchema @c @xs f
```

Reply:
I thought about it, but this seems to be quite specific to features, since it only makes sense when an `NP` is built out of distinct types. I'm not sure it's worth introducing a special abstraction just for this case.